src/qibo/quantum_info/quantum_networks.py

```diff
@@ -44,161 +45,227 @@ class QuantumNetwork:
-                object. Defaults to ``None``.
+        self.dims = reduce(mul, self.partition) if len(self.partition) > 0 else 1
 
-        Returns:
-            ndarray: Choi matrix of the quantum network.
+    @staticmethod
+    def _order_tensor2operator(n: int):
         """
-        if backend is None:  # pragma: no cover
-            backend = self._backend
+        Returns the order to reshape a tensor to an operator.
+        Given a tenosr of 2n leads, the order is [0, 2, 4, ..., 1, 3, 5, ...].
+        """
+        order = list(range(0, n * 2, 2)) + list(range(1, n * 2, 2))
+        return order
 
-        return backend.cast(self._matrix, dtype=self._matrix.dtype)
+    @staticmethod
+    def _order_operator2tensor(n: int):
+        """
+        Returns the order to reshape an operator to a tensor.
+        Given a operator of 2n systems, the order is [0, n, 1, n+1, 2, n+2, ...].
+        """
+        order = list(sum(zip(list(range(0, n)), list(range(n, n * 2))), ()))
+        return order
 
-    def is_pure(self):
-        """Returns bool indicading if the Choi operator of the network is pure."""
-        return self._pure
+    @classmethod
+    def _operator2tensor(cls, operator, partition: List[int]):
+        # check if tensorflow is installed
+        try:
+            from tensorflow import InvalidArgumentError
+        except ImportError:
+            InvalidArgumentError = ValueError
 
-    def is_hermitian(
-        self, order: Optional[Union[int, str]] = None, precision_tol: float = 1e-8
-    ):
-        """Returns bool indicating if the Choi operator :math:`\\mathcal{E}` of the network is Hermitian.
+        n = len(partition)
+        order = cls._order_operator2tensor(n)
 
-        Hermicity is calculated as distance between :math:`\\mathcal{E}` and
-        :math:`\\mathcal{E}^{\\dagger}` with respect to a given norm.
-        Default is the ``Hilbert-Schmidt`` norm (also known as ``Frobenius`` norm).
+        try:
+            tensor = np.transpose(operator.reshape(list(partition) * 2), order)
+            return tensor.reshape([dim**2 for dim in partition])
+        except ValueError:
+            raise_error(
+                ValueError,
+                "``partition`` does not match the shape of the input matrix. "
+                + f"Cannot reshape matrix of size {operator.shape} to partition {partition}",
+            )
+        except InvalidArgumentError:
+            raise_error(
+                ValueError,
+                "``partition`` does not match the shape of the input matrix. "
+                + f"Cannot reshape matrix of size {operator.shape} to partition {partition}",
+            )
+        except RuntimeError:
+            raise_error(
+                ValueError,
+                "``partition`` does not match the shape of the input matrix. "
+                + f"Cannot reshape matrix of size {operator.shape} to partition {partition}",
+            )
```

Comment:
There is a much simpler way of implementing this. Just check if `partition` matches `operator.shape`. If not, raise an error. If it does, the calculation is done. None of these `try / except` are needed.